clang/lib/Basic/CodeGenOptions.cpp

```diff
@@ -36,13 +36,13 @@ void CodeGenOptions::resetNonModularOptions(StringRef ModuleFormat) {
   // emitted into the PCM (-gmodules).
   if (ModuleFormat == "raw" && !DebugTypeExtRefs) {
 #define DEBUGOPT(Name, Bits, Default, Compatibility)                           \
-  if constexpr (CK::Compatibility == CK::Affecting)                            \
+  if constexpr (CK::Compatibility != CK::Benign)                               \
     Name = Default;
 #define VALUE_DEBUGOPT(Name, Bits, Default, Compatibility)                     \
-  if constexpr (CK::Compatibility == CK::Affecting)                            \
+  if constexpr (CK::Compatibility == CK::Benign)                               \
```

Comment:
This condition and the one below have been inverted from the original, while the one above is not.  But do we even need these `if`s? We already reset the benign options above, so resetting them again seems fine and less error prone than these conditions.

Reply:
Good catch. These should actually all be `!= CK::Benign`. Removing the condition would be fine semantically, but `if constexpr` is more efficient. I don't have a strong preference here, just wanted to keep the original behavior.